types/lodash/lodash-tests.ts

```diff
@@ -7389,3 +7407,152 @@ _.templateSettings; // $ExpectType TemplateSettings
+        type F = O3[number]; // $ExpectType "OK"
+
+        // And GetIndexedField<> should behave the same:
+        type G = GetIndexedField<O1, number>; // $ExpectType "OK"
+        type H = GetIndexedField<O2, number>; // $ExpectType "OK"
+        type I = GetIndexedField<O3, number>; // $ExpectType "OK"
+        type J = GetIndexedField<O3, 1>; // $ExpectType "OK"
+    }
+
+    {
+        type E = GetIndexedField<O1, bigint>; // $ExpectType undefined
+        type F = GetIndexedField<O2, boolean>; // $ExpectType undefined
+        type G = GetIndexedField<O3, null>; // $ExpectType undefined
+        type H = GetIndexedField<O1, undefined>; // $ExpectType undefined
+    }
+}
+
+// GetIndexedField: indexing strings
+
+{
+    type S1 = string;
+    type S2 = string & { x: 'OK' };
+
+    type A = GetIndexedField<S1, number>; // $ExpectType string
+    type B = GetIndexedField<S1, 1>; // $ExpectType string
+    type C = GetIndexedField<S1, '1'>; // $ExpectType string
+    type D = GetIndexedField<S1, 'length'>; // $ExpectType number
+    type F = GetIndexedField<S1, 'x'>; // $ExpectType undefined
+    type G = GetIndexedField<S2, 'x'>; // $ExpectType "OK"
+}
+
+// GetIndexedFields: indexing tuples
+
+{
+    type T1 = ['OK'];
+
+    type A = GetIndexedField<T1, number>; // $ExpectType "OK"
+    type B = GetIndexedField<T1, 0>; // $ExpectType "OK"
+    type C = GetIndexedField<T1, 1>; // $ExpectType undefined
+    type D = GetIndexedField<T1, '0'>; // $ExpectType "OK"
+    type E = GetIndexedField<T1, '1'>; // $ExpectType undefined
+    type F = GetIndexedField<T1, 'length'>; // $ExpectType 1
+}
+
+// GetFieldType: correctly handles any possible combination of dots and square
+// brackets in path type literals.
+{
+    interface O1 { key: 'value' };
+    interface O2 { some: { test: { path: 'value' } } };
+
+    interface O3 {
+        'some.key': 'value1';
+        'key[with]brackets': 'value2';
+    };
+
+    type A = GetFieldType<O1, 'key'>; // $ExpectType "value"
+    type B = GetFieldType<O1, '[key]'>; // $ExpectType "value"
+
+    type C = GetFieldType<O2, 'some.test.path'>; // $ExpectType "value"
+    type D = GetFieldType<O2, 'some.test[path]'>; // $ExpectType "value"
+    type E = GetFieldType<O2, 'some[test].path'>; // $ExpectType "value"
+    type F = GetFieldType<O2, '[some].test.path'>; // $ExpectType "value"
+    type G = GetFieldType<O2, 'some[test][path]'>; // $ExpectType "value"
+    type H = GetFieldType<O2, '[some].test[path]'>; // $ExpectType "value"
+    type I = GetFieldType<O2, '[some][test].path'>; // $ExpectType "value"
+    type J = GetFieldType<O2, '[some][test][path]'>; // $ExpectType "value"
+
+    // Here are tricky cases, where dots and square brackets inside key names
+    // mess-up with TypeScript parsing of path type literals we can do.
+
+    // No way to distinguish a single key with dot vs. two dot-separated keys,
+    // GetFieldType<> interpretes it as the later, thus undefined result.
```

Comment:
I'm curious if lodash even handles this correctly at runtime...

Reply:
Hopefully. If it does not, it is a bug to patch in `lodash`. I'll probably arrive to test it sometime later, after type resolution is perfected (any runtime fixes are sure easier to do).